src/view/MainViewManager.js

```diff
@@ -846,6 +847,62 @@ define(function (require, exports, module) {
     }
 
     /**
+     * swaps current views to opposite panes
+     */
+    function swapPaneContent() {
+
+        var activeFileInactiveView,
+            inactiveFileActiveView,
+            activePaneId = getActivePaneId(),
+            inactivePaneId = activePaneId === FIRST_PANE ? SECOND_PANE : FIRST_PANE,
+            activePane = _getPane(activePaneId),
+            inactivePane = _getPane(inactivePaneId),
+            activeFile = activePane.getCurrentlyViewedFile(),
+            inactiveFile = inactivePane.getCurrentlyViewedFile();
+
+        // Check if one of the panes is empty move and open currently viewed  file in opposite pane.
+        // if both are empty do nothing.
+        if (!activeFile || !inactiveFile) {
+            if (!activeFile && inactiveFile) {
+                _moveView(inactivePaneId, activePaneId, inactiveFile, 0);
+            } else if (!inactiveFile && activeFile) {
+                _moveView(activePaneId, inactivePaneId, activeFile, 0);
+            }
+
+        } else {
+
+            // check if currently viewed file is present in opposing pane.  If it is open the file.
+            // if it is not add it to the opposite pane and open.
+            activeFileInactiveView = inactivePane.getViewForPath(activeFile.fullPath);
+            inactiveFileActiveView = activePane.getViewForPath(inactiveFile.fullPath);
+
+            if (inactiveFileActiveView) {
```

Comment:
@petetnt I looked into moving most of what is now in MainViewManager onto the pane prototype (see below), and I can push up the commit including it, however, with this setup I can't seem to get the active pane to consistently stay in place.  It flips from one pane to the other every third swap. I've tried promises and a few other ideas, but Maybe I'm missing something obvious.  Any insights are welcome.   
```javascript
    Pane.prototype.swapPaneContent = function(oppositeFile) {
        var self = this,
            oppositePane = self.id === FIRST_PANE ? SECOND_PANE : FIRST_PANE;

        if(!oppositeFile){
            var file = self.getCurrentlyViewedFile();
            MainViewManager._moveView( self.id, oppositePane.id, file , 0);
        } else {
            if (self.getViewForPath(oppositeFile.fullPath)) {
                CommandManager.execute(Commands.FILE_OPEN, {
                    fullPath: oppositeFile.fullPath,
                    paneId: self.id
                });
            } else {
                CommandManager.execute(Commands.CMD_ADD_TO_WORKINGSET_AND_OPEN, {
                    fullPath: oppositeFile.fullPath,
                    paneId: self.id
                });
            };
        };
    };

```
```javascript
    function swapPaneContent() {

        var activePaneId = getActivePaneId(),
            inactivePaneId = activePaneId === FIRST_PANE ? SECOND_PANE : FIRST_PANE,
            activePane = _getPane(activePaneId),
            inactivePane = _getPane(inactivePaneId),
            activeFile = activePane.getCurrentlyViewedFile(),
            inactiveFile = inactivePane.getCurrentlyViewedFile();

            inactivePane.swapPaneContent(activeFile);   // addition 
            activePane.swapPaneContent(inactiveFile);  // addition
    };
```